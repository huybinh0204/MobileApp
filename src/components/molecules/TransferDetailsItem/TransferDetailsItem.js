import PropTypes from 'prop-types';
import React from 'react';
import {
  TransferDetailsItemContainer,
  TransferDetailsItemText,
} from './TransferDetailsItem.styles';

const TransferDetailsItem = ({ label, value, isLastItem }) => {
  return (
    <TransferDetailsItemContainer isLastItem={isLastItem}>
      <TransferDetailsItemText>{label}</TransferDetailsItemText>
      <TransferDetailsItemText>{value}</TransferDetailsItemText>
    </TransferDetailsItemContainer>
  );
};

TransferDetailsItem.defaultProps = {
  isLastItem: false,
};

TransferDetailsItem.propTypes = {
  isLastItem: PropTypes.bool,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

export default TransferDetailsItem;
