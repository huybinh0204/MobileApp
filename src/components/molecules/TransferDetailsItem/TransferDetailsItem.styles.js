import styled from '@emotion/native';
import { BaseText } from '_components/atoms';

export const TransferDetailsItemContainer = styled.View(({ theme, isLastItem }) => ({
  alignSelf: 'stretch',
  alignItems: 'center',
  flexDirection: 'row',
  justifyContent: 'space-between',
  borderBottomColor: isLastItem ? theme.colors.transparent : theme.colors.beta50,
  borderBottomWidth: 1,
  paddingVertical: theme.spacing.s,
  marginBottom: isLastItem ? theme.spacing.s : 0,
}));

export const TransferDetailsItemText = styled(BaseText)(({ theme }) => ({
  ...theme.typography.tinyContentRegular,
  color: theme.colors.beta900,
}));
