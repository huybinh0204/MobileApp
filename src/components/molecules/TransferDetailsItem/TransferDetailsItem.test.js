import { render } from '@testing-library/react-native';
import React from 'react';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import TransferDetailsItem from './TransferDetailsItem';

describe('TransferDetailsItem', () => {
  let props;

  beforeEach(() => {
    props = {
      label: 'Label',
      value: 'Value',
    };
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders component={TransferDetailsItem} componentProps={props} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
