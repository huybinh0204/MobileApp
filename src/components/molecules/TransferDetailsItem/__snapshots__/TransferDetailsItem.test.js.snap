// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferDetailsItem should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        isLastItem={false}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "borderBottomColor": "#F2F3F4",
                              "borderBottomWidth": 1,
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "marginBottom": 0,
                              "paddingVertical": 29,
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#1F1D25",
                                "fontFamily": "Poppins",
                                "fontSize": 21.5,
                                "lineHeight": 30.5,
                              },
                              undefined,
                            ]
                          }
                        >
                          Label
                        </Text>
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#1F1D25",
                                "fontFamily": "Poppins",
                                "fontSize": 21.5,
                                "lineHeight": 30.5,
                              },
                              undefined,
                            ]
                          }
                        >
                          Value
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
